QueryEngine/NativeCodegen.cpp

```diff
@@ -2875,6 +3030,29 @@ Executor::compileWorkUnit(const std::vector<InputTableInfo>& query_infos,
   }
 
   // Generate final native code from the LLVM IR.
+  std::shared_ptr<CompilationContext> compilation_context;
+  switch (co.device_type) {
+    case ExecutorDeviceType::CPU:
+      compilation_context =
+          optimizeAndCodegenCPU(query_func, multifrag_query_func, live_funcs, co);
+      break;
+    case ExecutorDeviceType::GPU:
+      compilation_context = optimizeAndCodegenGPU(query_func,
+                                                  multifrag_query_func,
+                                                  live_funcs,
+                                                  is_group_by || ra_exe_unit.estimator,
+                                                  cuda_mgr,
+                                                  co);
+      break;
+    case ExecutorDeviceType::L0:
+      compilation_context = optimizeAndCodegenL0(
+          query_func, multifrag_query_func, live_funcs, false, l0_mgr, co);
+      break;
+
+    default:
+      LOG(FATAL) << "Invalid device type";
+      return {};
+  }
   return std::make_tuple(
       CompilationResult{
           co.device_type == ExecutorDeviceType::CPU
```

Comment:
Looks like a merge bug, optimizeAndCodegen is called twice

Reply:
Fixed